### DISCUSSION TITLE
How to persist repeatable jobs when the queue is paused

### DISCUSSION THREAD

**Discussion by Atralbus** [OptimalBits/bull#2614] [Q&A]
I am using Bull with repeatable jobs. I want to be able to pause the queue while I'm updating the server. I was planning to use pause method of Bull but what I realized is that once the queue is unpaused, Bull only runs the repeatable jobs that was registered before the pause. So if there were some jobs that were supposed to run while the queue is paused they are just being ignored.

For example, I add a job in queue at 16:52 with repeat definition which triggers the job every 10 minutes. If I pause the queue at 16:55 and unpause it at 17:25, it only runs the job that was registered to the queue before the pause (the one that was supposed to run at 17:00). But then it runs the next job at 17:30 and completely ignores the jobs that were supposed to run at 17:10 and 17:20.

I guess this behavior is fine for most cases but in my case jobs collect some data based on a time range and scheduled time. For example, if a job is scheduled to 17:00, it collects data created between 16:50-17:00. So I want to be able to re-run all the missed jobs while the queue is paused.

I appreciate any insights or suggestions on how to handle this situation effectively. Thank you!

**Comment by manast** [comment]
The way repeatable jobs work is by placing a new delayed job as soon as a job is processed, therefore while you are paused this whole mechanism is also paused. I am afraid there is no current solution for handling this use case, although I recognize it is a legit use case.

**Reply by Atralbus** [reply]
Yes, I inspected redis db to see how the jobs were being created. It looked like each job was creating the record for the next job. I was wondering if this could be an easy solution considering how Bull works. Each repeatable job is aware of the repeat options, so instead of creating a job for the next possible future date, would it be easy to create the next job based on repeat options until it reaches to current datetime. Similar to `isLocal` and `doNotWaitActive` flags, this behavior could be selected with some other flag.

If you think this behavior is not in the scope of Bull, do you have any suggestion on how to solve this problem? The easiest solution I could think of is to check all the jobs and figure out which jobs were supposed the run while the server was in maintenance. But it doesn't sound convenient.

**Reply by manast** [reply]
Since the queue is paused, there is no mechanism running that could add the next iteration of any repeatable job, so there is no easy work around AFAIK.
